metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/parser.py

```diff
@@ -0,0 +1,64 @@
+import importlib.resources as pkg_resource
+import logging
+from typing import List, cast
+
+import lark
+from lark import Lark, Tree
+
+from datahub.ingestion.source.powerbi.config import PowerBiDashboardSourceReport
+from datahub.ingestion.source.powerbi.m_query import resolver, validator
+from datahub.ingestion.source.powerbi.proxy import PowerBiAPI
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _parse_expression(expression: str) -> Tree:
+    # Read lexical grammar as text
+    grammar: str = pkg_resource.read_text(
+        "datahub.ingestion.source.powerbi", "powerbi-lexical-grammar.rule"
+    )
+
+    # Create lark parser for the grammar text
+    lark_parser = Lark(grammar, start="let_expression", regex=True)
```

Comment:
Can we save this as an higher level global variable? My guess is that creating the parser is not a cheap process?

Reply:
Done